research/gnn-survey/layers.py

```diff
@@ -167,12 +167,14 @@ def call(self, inputs):
     attn_row = tf.matmul(x, self.attention_row)
     attn_col = tf.matmul(x, self.attention_col)
 
-    attn = tf.sparse.add(adj * attn_row,
+    attn = tf.sparse.add(adj * attn_row, 
                          adj * tf.transpose(attn_col, perm=[1, 0]))
     attn = tf.sparse.SparseTensor(
         indices=attn.indices,
         values=self.leakyrelu(attn.values),
         dense_shape=attn.dense_shape)
+
+    attn = tf.sparse.reorder(attn)
```

Comment:
This shouldn't be necessary since the previous operation doesn't manipulate the indices?

Reply:
I'm not very sure but I think this may help sparse.softmax function below apply correctly. Since the indices have the wrong dimension at first and the adj matrix is not symmetric, I think that it may be added?(Let softmax apply on write dimension, and acc also improve a lot with reorder function)

Reply:
Okay, add just add a comment that says that, then.

Reply:
OK!